### PR TITLE
Dev/81 [POST]/exams/{examId}/groups

### PR DESCRIPTION
**Pull request by Chaoyuuu**

### CODE REVIEW

File: src/app/exam/participants/Examinees.js [hunk 1]
```diff
@@ -23,16 +22,19 @@ const Examinees = () => {
...
+    }
+
     useEffect(() => {
         if (!currentExam) {
             refetchExam(examId)
         }
         if (!examinees && currentExam) {
-            examService.getExaminees(examId)
-                .then(students => setExaminees(students));
+            fetchExaminees(examId)
```
Comment (Wally5077) [inline src/app/exam/participants/Examinees.js:37]: 請補上分號

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: src/app/exam/participants/Examinees.js [hunk 2]
```diff
@@ -142,7 +134,8 @@ const Examinees = () => {
                                    buttonName: "Add"
                                }}
                                show={showAddGroupModal}
-                               onClose={() => setShowAddGroupModal(false)}/>
+                               onClose={() => setShowAddGroupModal(false)}
+                               onSubmit={groupNames => addExamineesByGroups(groupNames)}/>
```
Comment (nyngwang) [inline src/app/exam/participants/Examinees.js:138]: Can write `onSubmit={addExamineesByGroups}`.

File: src/app/exam/participants/Examinees.js [hunk 3]
```diff
@@ -131,7 +123,7 @@ const Examinees = () => {
                                }}
                                show={showAddStudentModal}
                                onClose={() => setShowAddStudentModal(false)}
-                               onSubmit={emails => addExaminees(emails)}/>
+                               onSubmit={emails => addExamineesByEmails(emails)}/>
```
Comment (nyngwang) [inline src/app/exam/participants/Examinees.js:126]: ditto

File: src/app/exam/participants/Examinees.js [hunk 4]
```diff
@@ -48,34 +50,24 @@ const Examinees = () => {
...
+        fetchExaminees(examId);
     }
 
-    const removeExaminee = (email) => {
-        examService.deleteExaminees(examId, email)
-            .then(() => removeExamineeByEmail(email));
+    const addExamineesByGroups = async (groupNames) => {
+        await examService.addGroupsOfExaminees(examId, groupNames)
+            .then(res => console.log("res in addGroupsOfExaminees", res));
+        fetchExaminees(examId);
```
Comment (nyngwang) [inline src/app/exam/participants/Examinees.js:65]: No need to async, just put `fetchExaminees(examId)` into `.then()`.

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review Done.

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by nyngwang — APPROVED** [review]
LGTM

File: src/app/exam/participants/Examinees.js [hunk 5]
```diff
@@ -23,16 +22,19 @@ const Examinees = () => {
     const [selectedExaminee, setSelectedExaminee] = useState(undefined);
     const [showAddStudentModal, setShowAddStudentModal] = useState(false);
     const [showAddGroupModal, setShowAddGroupModal] = useState(false);
-
     const [showRemoveExamineeConfirmationModal, setShowRemoveExamineeConfirmationModal] = useState(false);
 
+    const fetchExaminees = (examId) => {
+        examService.getExaminees(examId)
+            .then(students => setExaminees(students));
```
[the remark this answers is not in the document]
Comment (Johnny850807) [reply src/app/exam/participants/Examinees.js:29]: rename to `examineeStudents`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**